fdbrpc/TokenSign.cpp

```diff
@@ -117,6 +117,71 @@ bool verifyToken(SignedAuthTokenRef signedToken, StringRef publicKeyDer) {
 	return true;
 }
 
+StringRef hmacSHA256Sign(Arena& arena, StringRef str, StringRef key) {
+	unsigned int mdLength = 256 / 8;
+	auto md = new (arena) uint8_t[mdLength];
+	::HMAC(::EVP_sha256(), key.begin(), key.size(), str.begin(), str.size(), md, &mdLength);
+	return StringRef(md, mdLength);
+}
+
+bool hmacSHA256Verify(StringRef str, StringRef signature, StringRef key) {
+	Arena arena;
+	auto res = hmacSHA256Sign(arena, str, key);
+	return signature == res;
+}
+
+enum class CryptAlgo { ES256, HMACSHA256 };
+
+StringRef sign(CryptAlgo algo, Arena arena, StringRef str, StringRef key) {
```

Comment:
This should be pass by reference.

Reply:
oh good catch -- this is a memory bug, will fix